### PR TITLE
Add support to customize form buttons text

### PR DESCRIPTION
**Pull request by Kvaz1r**
### Description:

Form is very convenient widget but sometimes there is need to change button labels so I propose to add such feature. 

### Checklist:

- [X] Tests included.
- [ ] Lint and formatter run with no errors.
- [X] Tests all pass.

#### Where applicable:

- [] Public APIs match existing style.
- [X] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

**Comment by Kvaz1r** [comment]
Right, fixed.

File: widget/form.go [hunk 1]
```diff
@@ -82,13 +84,18 @@ func (f *Form) CreateRenderer() fyne.WidgetRenderer {
...
-		buttons.Append(NewButtonWithIcon("Cancel", theme.CancelIcon(), f.OnCancel))
+		if f.CancelButton == nil {
+			f.CancelButton = NewButtonWithIcon("Cancel", theme.CancelIcon(), f.OnCancel)
+		}
+		buttons.Append(f.CancelButton)
 	}
 	if f.OnSubmit != nil {
-		submit := NewButtonWithIcon("Submit", theme.ConfirmIcon(), f.OnSubmit)
-		submit.Style = PrimaryButton
+		if f.CancelButton == nil {
```
Comment (andydotxyz) [inline widget/form.go]: I think this is a copy / paste bug.

File: widget/form.go [hunk 2]
```diff
@@ -24,9 +24,11 @@ func NewFormItem(text string, widget fyne.CanvasObject) *FormItem {
 type Form struct {
 	BaseWidget
 
-	Items    []*FormItem
-	OnSubmit func()
-	OnCancel func()
+	Items        []*FormItem
+	SubmitButton *Button
```
Comment (andydotxyz) [inline widget/form.go]: I think your idea of setting text worked more in-line with how other widgets are structured.
That way we control the internal details of how the buttons look and behave, should provide a more consistent UI overall.
And shorter code for people wanting to change the text :)
Apologies that this was not clear the way the previous review was worded.

Comment (Kvaz1r) [reply widget/form.go]: Fixed.

File: widget/form_test.go [hunk 3]
```diff
@@ -18,7 +17,7 @@ func TestFormSize(t *testing.T) {
 
 func TestForm_CreateRenderer(t *testing.T) {
 	form := &Form{Items: []*FormItem{{Text: "test1", Widget: NewEntry()}}}
-	assert.NotNil(t, test.WidgetRenderer(form))
+	assert.NotNil(t, Renderer(form))
```
Comment (andydotxyz) [inline widget/form_test.go]: This is great, thanks. Ready to merge but then I noticed this.
The change goes back to using a deprecated method from the new test one that looks up correctly...

Comment (Kvaz1r) [reply widget/form_test.go]: Well it's automatic goimports change that used by editor on saving, I don't know why it happens =)
I'll back it by hand.

Comment (Kvaz1r) [reply widget/form_test.go]: Ah, I got it. I made first changes on master branch and only after that switched to develop branch but forget properly update file. Fixed now.

**Comment by andydotxyz** [comment]
Thanks for adding this :)

**Pull request merged**